stm32/Src/main.c

```diff
@@ -152,8 +167,14 @@ void SystemClock_Config(void)
   RCC_OscInitStruct.LSEState = RCC_LSE_ON;
   RCC_OscInitStruct.MSIState = RCC_MSI_ON;
   RCC_OscInitStruct.MSICalibrationValue = RCC_MSICALIBRATION_DEFAULT;
-  RCC_OscInitStruct.MSIClockRange = RCC_MSIRANGE_11;
-  RCC_OscInitStruct.PLL.PLLState = RCC_PLL_NONE;
+  RCC_OscInitStruct.MSIClockRange = RCC_MSIRANGE_6;

```

Comment:
Why is the RTC clock changed? This may cause issues with the LoRa libraries since it requires a specific clock configuration so that it will run while the chip is in STOP2 mode. Have you checked it still works?